core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -44,9 +37,9 @@ object ReplicaAssignment {
  * @param addingReplicas the replicas that are being added if there is a pending reassignment
  * @param removingReplicas the replicas that are being removed if there is a pending reassignment
  */
-case class ReplicaAssignment(replicas: Seq[Int],
-                             addingReplicas: Seq[Int],
-                             removingReplicas: Seq[Int]) {
+case class ReplicaAssignment private (replicas: Seq[Int],
```

Comment:
nit: is this conventional? I don't think we use this ordering anywhere else in the code?

Reply:
Are you referring to the `private` keyword after `ReplicaAssignment`? This is how to declare the "default" constructor private in Scala. This is the best reference I could find: https://www.scala-lang.org/download/changelog.html#access-modifiers-for-primary-constructors

Nicely we never create this class directly (outside of JSON to ReplicaAssignment decoding). We either use some of the other `apply` methods or the `reassignTo` method. I don't think in the future we will want to create the value directly so I thought might as well make it private.

Reply:
Ah, ok. First time I have seen this.